### PR TITLE
Allow custom activation in SqueezeExcitation of EfficientNet

### PR DESCRIPTION
**Pull request by kazhang**
It would be great to reuse SqueezeExcitation in RegNet implementation(#4403).

cc @datumbox

### CODE REVIEW

File: torchvision/models/efficientnet.py [hunk 1]
```diff
@@ -32,15 +32,23 @@
...
+        self,
+        input_channels: int,
+        squeeze_channels: int,
+        activation: Optional[Callable[..., nn.Module]] = None,
+    ) -> None:
         super().__init__()
         self.fc1 = nn.Conv2d(input_channels, squeeze_channels, 1)
         self.fc2 = nn.Conv2d(squeeze_channels, input_channels, 1)
+        if activation is None:
+            activation = nn.SiLU
```
Comment (datumbox) [inline torchvision/models/efficientnet.py:45]: Given the original Squeeze-and-Excitation [paper](https://arxiv.org/pdf/1709.01507.pdf) uses ReLU, perhaps that's a better default option for when we move it out to the `ops` namespace?

File: torchvision/models/efficientnet.py [hunk 2]
```diff
@@ -32,15 +32,23 @@
...
+        self.activation = activation(inplace=True)
```
Comment (datumbox) [inline torchvision/models/efficientnet.py:46]: I would probably drop `inplace` from here because not all activation methods in PyTorch support the attribute (for example `Sigmoid`).

File: torchvision/models/efficientnet.py [hunk 3]
```diff
@@ -32,15 +32,23 @@
...
 
     def _scale(self, input: Tensor) -> Tensor:
         scale = F.adaptive_avg_pool2d(input, 1)
         scale = self.fc1(scale)
-        scale = F.silu(scale, inplace=True)
+        scale = self.activation(scale)
         scale = self.fc2(scale)
         return scale.sigmoid()
```
Comment (datumbox) [inline torchvision/models/efficientnet.py:53]: [MobileNetV3](https://github.com/pytorch/vision/blob/main/torchvision/models/mobilenetv3.py#L35) uses `hardsigmoid` here, so we probably need another option to allow someone replacing sigmoid with another activation.

File: torchvision/models/efficientnet.py [hunk 4]
```diff
@@ -32,15 +32,23 @@
 
 
 class SqueezeExcitation(nn.Module):
```
Comment (datumbox) [inline torchvision/models/efficientnet.py:34]: I'm still on the fence on whether we should mark this class with `_` to explicitly denote it's private (I believe that's what you did on your other PR) or leave as is for when we move it to `ops`.  Thoughts?

Would you consider making this PR replace ALL existing SE layers? If yes, we could move this straight to `ops`. Happy to help with converting existing models in a BC manner.

**Review by datumbox — COMMENTED** [review]
Thanks for the PR @kazhang. This is much needed and will allow us to reuse the layer in multiple places. I left a couple of comments, let me know what you think.

File: torchvision/models/efficientnet.py (hunk 4 above)
Comment (kazhang) [reply torchvision/models/efficientnet.py:34]: > I'm still on the fence on whether we should mark this class with _ to explicitly denote it's private
I'm considering to make it public since it has been published for a while and still popular (https://paperswithcode.com/method/squeeze-and-excitation-block)

 > Would you consider making this PR replace ALL existing SE layers? If yes, we could move this straight to ops. Happy to help with converting existing models in a BC manner.
I would like to get the consensus in #4333 first and execute carefully. I'm thinking to merge this PR first to avoid the third SE implementation and then consolidate the two into one.

**Review by datumbox — APPROVED** [review]
LGTM, thanks @kazhang!
